PhotoAndCopyShop.ReceiverService/ReceiverDaemon.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Messaging.ServiceBus;
+
+namespace PhotoAndCopyShop.ReceiverService;
+
+public class ReceiverDaemon
+{
+    private readonly string _connectionString;
+    private readonly string _queueName;
+    private readonly string _targetFolder;
+    private readonly HashSet<string> _receivedMessages = new();
+
+    public ReceiverDaemon(string connectionString, string queueName, string targetFolder)
+    {
+        _connectionString = connectionString;
+        _queueName = queueName;
+        _targetFolder = targetFolder;
+    }
+
+    public async Task StartReceivingJobAsync()
```

Comment:
This method looks excessively big and contains multiple "jump" operators like return and continue. Can you decompose it to smaller parts?

Reply:
extracted file-related write operations into SaveMessageData()